### DISCUSSION TITLE
Problems with RLS

### DISCUSSION THREAD

**Discussion by TuurDutoit** [phil-opp/blog_os#630] [General]
My project is building fine with `cargo xbuild` as per [the instructions](https://os.phil-opp.com/minimal-rust-kernel/#building-our-kernel). The Rust extension for VSCode - which uses the Rust Language Server - is having a hard time compiling it however, but I managed to fix that so I wanted to document how I did that, in case it can help anyone else.

The initial issue, about the `panic_impl` being defined twice, was fixed with [these instructions](https://github.com/rust-lang/rls/issues/904#issuecomment-429456389) I found. With that fixed, another issue popped up about the `core` crate not being found. This is because the RLS is not using `xbuild`. However, a workaround is possible because `xbuild` places the compilation output in the `target` directory, where a normal Cargo build can find it. I added [this comment](https://github.com/rust-lang/rls/issues/904#issuecomment-506640850) to the issue above detailing how to go about it.

**Comment by phil-opp** [comment]
Thank you for sharing your solution! Another way that should also work is to set the `RUSTFLAGS` environment variable when launching the editor, e.g. `RUSTFLAGS="--sysroot /home/…/target/sysroot" code .` for VSCode. Then everything should work as normal. (See https://twitter.com/phil_opp/status/1143839524214190080.)

**Comment by phil-opp** [comment]
I hope that https://github.com/rust-lang/rfcs/pull/2663 makes progess soon, so that the whole `cargo-xbuild`/`xargo` wrapper is no longer needed. This should make RLS and rust-analyzer work out of the box, too.
